lib/replicator.js

```diff
@@ -1838,15 +1838,20 @@ module.exports = class Replicator {
 
     let clear = null
 
+    const reader = this.core.storage.createReadBatch()
+
+    const blocks = new Map()
     for (const b of this._blocks) {
-      if (this.core.bitfield.get(b.index) === false) continue
+      blocks.set(b.index, this.core.blocks.get(reader, b.index))
+    }
 
-      const reader = this.core.storage.createReadBatch()
+    reader.tryFlush()
+
+    for (const b of this._blocks) {
```

Comment:
Sanity check: this shouldn't be `for (const b of blocks) {` ? Otherwise it's iterating over a data structure than can mutate during iteration, so line 1851 can have pretty strange behaviour if the entry doesn't exist.